### PR TITLE
[inductor] Make clone_graph copy node name as well

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103557
* __->__ #103409

Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov @chenyang78

### CODE REVIEW

File: torch/_inductor/pattern_matcher.py [hunk 1]
```diff
@@ -1046,6 +1046,7 @@ def run_node(self, old_node):
             new_node = super().run_node(old_node)
             if isinstance(new_node, torch.fx.Proxy):
                 new_node.node.meta.update(old_node.meta)
+                new_node.node.name = old_node.name
```
Comment (eellison) [inline torch/_inductor/pattern_matcher.py:1049]: I think you should use this API instead to safe  https://github.com/pytorch/pytorch/blob/main/torch/fx/graph.py#L840

**Review by eellison — COMMENTED** [review]
i think we want to use the existing name api in case there is a conflict

**Review by eellison — APPROVED** [review]
looks good with api change

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**